Fix coloring of BitmapText with overlapping attributes

phantom10111 opened the pull request:
When `AddAttribute()` is called multiple times with overlapping ranges, old attributes need to be removed and/or fixed. Otherwise the old colors might show up when the text is rendered.

teejusb commented on src/BitmapText.cpp:
```diff
@@ -911,7 +911,50 @@ void BitmapText::AddAttribute( std::size_t iPos, const Attribute &attr )
+	const std::size_t iEndPos = iStartPos + newAttr.length;
+
+	// First attribute starting at the same position or further than the new attribute
+	std::map<std::size_t, Attribute>::iterator iterFirstAfterStart = m_mAttributes.lower_bound( iStartPos );
```
It's totally okay to use `auto` when defining the types for these iterators. Can we make that change here and below?

phantom10111 committed:
Fix coloring of BitmapText with overlapping attributes

When AddAttribute() is called multiple times with overlapping ranges, old
attributes need to be removed and/or fixed. Otherwise the old colors might
show up when the text is rendered.

teejusb commented:
TYSM! LGTM!
